Lib/ntpath.py

```diff
@@ -311,6 +311,40 @@ def ismount(path):
         return False
 
 
+_reserved_chars = frozenset(
+    {chr(i) for i in range(32)} |
+    {'"', '*', ':', '<', '>', '?', '|'}
+)
+
+_reserved_names = frozenset(
+    {'CON', 'PRN', 'AUX', 'NUL', 'CONIN$', 'CONOUT$'} |
+    {f'COM{c}' for c in '123456789\xb9\xb2\xb3'} |
+    {f'LPT{c}' for c in '123456789\xb9\xb2\xb3'}
+)
+
+def isreserved(path):
+    """Return true if the pathname is reserved by the system."""
+    # Refer to "Naming Files, Paths, and Namespaces":
+    # https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file
+    path = os.fsdecode(splitdrive(path)[1]).rstrip(r'\/')
+    while True:
+        path, name = split(path)
```

Comment:
If we had a `_split_internal()` function, we could avoid calling `splitdrive()` repeatedly. For example:

```python
def split(p):
    """Split a pathname.

    Return tuple (head, tail) where tail is everything after the final slash.
    Either part may be empty."""
    d, p = splitdrive(p)
    head, tail = _split_internal(p)
    return d + head, tail


def _split_internal(p):
    seps = _get_bothseps(p)
    # set i to index beyond p's last slash
    i = len(p)
    while i and p[i-1] not in seps:
        i -= 1
    head, tail = p[:i], p[i:]  # now tail has no slashes
    # remove trailing slashes from head, unless it's all slashes
    head = head.rstrip(seps) or head
    return head, tail
```
```python
def isreserved(path):
    """Return true if the pathname is reserved by the system."""
    # Refer to "Naming Files, Paths, and Namespaces":
    # https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file
    path = os.fsdecode(splitdrive(path)[1]).rstrip(r'\/')
    while True:
        path, name = _split_internal(path)
        if not name:
            return False
    ...
```

Reply:
I've pushed a different solution which calls `path.split(sep)` rather than `split(path)`. My concern with your suggestion is that it complicates `split()` a little.